### PR TITLE
Add third-party tests for BeanieODM

### PR DESCRIPTION
**Pull request by mdaffad**
## Change Summary

Integrate beanie tests on third party test workflow, part of third party tests on #11160.
Workflow result - https://github.com/mdaffad/pydantic/actions/runs/12477146497

## Related issue number

#11160

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: .github/workflows/third-party.yml [hunk 1]
```diff
@@ -113,18 +113,82 @@ jobs:
...
+
+    - uses: astral-sh/setup-uv@v4
+      with:
+        enable-cache: true
+        python-version: ${{ matrix.python-version }}
+
+    - name: Start MongoDB
+      uses: supercharge/mongodb-github-action@1.11.0
+      with:
+        mongodb-version: ${{ matrix.mongodb-version }}
```
Comment (Viicos) [inline .github/workflows/third-party.yml:149]:
```suggestion
        mongodb-version: '8.0'
```

Let's test against a single MongoDB version.

File: .github/workflows/third-party.yml [hunk 2]
```diff
@@ -113,18 +113,82 @@ jobs:
...
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'schedule' && github.repository == 'pydantic/pydantic') ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'third-party-tests'))
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        mongodb-version: [4.4, 5.0, 6.0, 7.0, 8.0 ]
```
Comment (Viicos) [inline .github/workflows/third-party.yml:129]:
```suggestion
```

File: .github/workflows/third-party.yml [hunk 3]
```diff
@@ -113,18 +113,82 @@ jobs:
...
+
+    - name: Checkout Pydantic
+      uses: actions/checkout@v4
+      with:
+        path: pydantic-latest
+
+    - uses: astral-sh/setup-uv@v4
+      with:
+        enable-cache: true
+        python-version: ${{ matrix.python-version }}
```
Comment (Viicos) [inline .github/workflows/third-party.yml:144]:
```suggestion
    - uses: actions/setup-python@v5
      with:
        python-version: ${{ matrix.python-version }}
```

Best to be as close to the project's CI.

File: .github/workflows/third-party.yml [hunk 4]
```diff
@@ -113,18 +113,82 @@ jobs:
...
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'third-party-tests'))
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        mongodb-version: [4.4, 5.0, 6.0, 7.0, 8.0 ]
+    steps:
+    - name: Checkout beanie
```
Comment (Viicos) [inline .github/workflows/third-party.yml:131]:
```suggestion
    - name: Checkout Beanie
```

File: .github/workflows/third-party.yml [hunk 5]
```diff
@@ -113,18 +113,82 @@ jobs:
...
+      uses: supercharge/mongodb-github-action@1.11.0
+      with:
+        mongodb-version: ${{ matrix.mongodb-version }}
+        mongodb-replica-set: test-rs
+
+    - name: Install beanie dependencies
+      run: |
+        uv venv --python ${{ matrix.python-version }}
+        uv pip install .[test,ci]
+        uv pip install ./pydantic-latest
```
Comment (Viicos) [inline .github/workflows/third-party.yml:156]:
```suggestion
    - name: Install Beanie dependencies
      run: |
        pip install .[test,ci]
        pip uninstall --yes pydantic 
        pip install ./pydantic-latest
```

File: .github/workflows/third-party.yml [hunk 6]
```diff
@@ -113,18 +113,82 @@ jobs:
...
+
+    - name: List installed dependencies
+      run: uv pip list
+
+    - name: Run beanie tests
+      env:
+        PYTHON_JIT: 1
```
Comment (Viicos) [inline .github/workflows/third-party.yml:163]:
```suggestion
    - name: Run Beanie tests
```

**Review by Viicos — CHANGES_REQUESTED** [review]
Thanks for the PR. This looks great

File: .github/workflows/third-party.yml [hunk 7]
```diff
@@ -113,6 +113,64 @@ jobs:
...
+    - name: Checkout Beanie
+      uses: actions/checkout@v4
+      with:
+        repository: BeanieODM/beanie
+
+    - name: Checkout Pydantic
+      uses: actions/checkout@v4
+      with:
+        path: pydantic-latest
+
```
Comment (sydney-runkle) [inline .github/workflows/third-party.yml:139]:
```suggestion

```

**Comment by sydney-runkle** [comment]
LGTM, let's wait for @Viicos' approval before we merge :)

**Review by Viicos — APPROVED** [review]
Thanks!

**Pull request merged**